File: mobile-app/lib/models/ai_workflow.dart
class AiPlanStep {
  final String agent;
  final String task;
  final String status;

  AiPlanStep({required this.agent, required this.task, required this.status});

  factory AiPlanStep.fromJson(Map<String, dynamic> json) => AiPlanStep(
        agent: json['agent'] as String? ?? '',
        task: json['task'] as String? ?? '',
        status: json['status'] as String? ?? 'Pending',
      );
}

class AiWorkflow {
  final String id;
  final String patientProfileId;
  final String objective;
  final String? summary;
  final List<AiPlanStep> steps;
  final String status;
  final String? errorMessage;
  final String modelUsed;
  final String toolCallSummary;
  final String reviewStatus;
  final String? reviewedByName;
  final String? reviewNotes;
  final DateTime createdAt;

  AiWorkflow({
    required this.id,
    required this.patientProfileId,
    required this.objective,
    this.summary,
    required this.steps,
    required this.status,
    this.errorMessage,
    required this.modelUsed,
    required this.toolCallSummary,
    required this.reviewStatus,
    this.reviewedByName,
    this.reviewNotes,
    required this.createdAt,
  });

  factory AiWorkflow.fromJson(Map<String, dynamic> json) => AiWorkflow(
        id: json['id'] as String,
        patientProfileId: json['patientProfileId'] as String? ?? '',
        objective: json['objective'] as String? ?? '',
        summary: json['summary'] as String?,
        steps: ((json['steps'] as List<dynamic>?) ?? [])
            .map((e) => AiPlanStep.fromJson(e as Map<String, dynamic>))
            .toList(),
        status: json['status'] as String? ?? '',
        errorMessage: json['errorMessage'] as String?,
        modelUsed: json['modelUsed'] as String? ?? '',
        toolCallSummary: json['toolCallSummary'] as String? ?? '',
        reviewStatus: json['reviewStatus'] as String? ?? 'NotReviewed',
        reviewedByName: json['reviewedByName'] as String?,
        reviewNotes: json['reviewNotes'] as String?,
        createdAt: DateTime.parse(json['createdAt'] as String),
      );
}

File: backend-api/Entities/Ai/AiWorkflow.cs
namespace CarePulse.Api.Entities.Ai;

public enum AiWorkflowStatus
{
    PlanCreated,
    ValidationFailed,
    LlmError
}

public enum AiPlanReviewStatus
{
    NotReviewed,
    Approved,
    Rejected
}

/// <summary>
/// One record per Agent 1 (Planner/Coordinator) invocation. Immutable audit-style
/// record — intentionally does NOT inherit BaseEntity (no soft delete), matching
/// EmergencyAlertLog: a workflow run must never be hidden or edited, only reviewed.
/// </summary>
public class AiWorkflow
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid PatientProfileId { get; set; }

    public string Objective { get; set; } = string.Empty;

    /// <summary>Snapshot of the PatientContextDto passed to the LLM, for auditability.</summary>
    public string ContextSnapshotJson { get; set; } = "{}";

    /// <summary>Records the one allow-listed tool call Agent 1 made (name, timing, result summary).</summary>
    public string ToolCallSummary { get; set; } = string.Empty;

    public string? PlanSummary { get; set; }

    /// <summary>JSON array of {"agent","task","status"} — the delegated steps.</summary>
    public string StepsJson { get; set; } = "[]";

    public AiWorkflowStatus Status { get; set; }
    public string? ErrorMessage { get; set; }
    public string ModelUsed { get; set; } = string.Empty;

    public string CreatedByUserId { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public AiPlanReviewStatus ReviewStatus { get; set; } = AiPlanReviewStatus.NotReviewed;
    public string? ReviewedByUserId { get; set; }
    public DateTime? ReviewedAt { get; set; }
    public string? ReviewNotes { get; set; }
}
